docs/getting-started/backends-and-brokers/rabbitmq.rst

```diff
@@ -172,3 +172,62 @@ but rather use the :command:`rabbitmqctl` command:
     $ sudo rabbitmqctl stop
 
 When the server is running, you can continue reading `Setting up RabbitMQ`_.
+
+.. _using-quorum-queues:
+
+Using Quorum Queues
+===================
+
+.. warning::
+
+    Quorum Queues require disabling global QoS which means some features won't work as expected.
+    See `limitations`_ for details.
+
+Celery supports `Quorum Queues`_ by setting the ``x-queue-type`` header to ``quorum` like so:
+
+.. code-block:: python
+
+    from kombu import Queue
+
+    task_queues = [Queue('my-queue', queue_arguments={'x-queue-type': 'quorum'})]
+    broker_transport_options = {"confirm_publish": True}
+
+If you'd like to change the type of the default queue, set the :setting:`task_default_queue_type` setting to ``quorum``.
+
+Celery automatically detects if quorum queues are used using the :setting:`worker_detect_quorum_queues` setting.
+We recommend to keep the default behavior turned on.
+
+.. _`Quorum Queues`: https://www.rabbitmq.com/docs/quorum-queues
+
+.. _limitations:
+
+Limitations
+-----------
+
+Disabling global QoS means that the the per-channel QoS is now static.
+This means that some Celery features won't work when using Quorum Queues.
+
+Autoscaling relies on increasing and decreasing the prefetch count whenever a new process is instantiated
+or terminated so it won't work when Quorum Queues are detected.
+
+Similarly, the :setting:`worker_enable_prefetch_count_reduction` setting will be a no-op even when set to ``True``
+when Quorum Queues are detected.
+
+In addition, :ref:`ETA/Countdown <calling-eta>` will block the worker when received until the ETA arrives since
+we can no longer increase the prefetch count and fetch another task from the queue.
+
+In order to properly schedule ETA/Countdown tasks you need to enable :ref:`Native Delayed Delivery <native-delayed-delivery>`.
+
+.. _native-delayed-delivery:
+
+Native Delayed Delivery
+-----------------------
+
+Since tasks with ETA/Countdown will block the worker until they are scheduled for execution,
+we need to use RabbitMQ's native capabilities to schedule the execution of tasks.
+
+The design is borrowed from NServiceBus. If you are interested in the implementation details, refer to their `documentation`_.
+
+.. _documentation: https://docs.particular.net/transports/rabbitmq/delayed-delivery
+
+To enable Native Delayed Delivery set the :setting:`broker_native_delayed_delivery` setting to ``True``.
```

Comment:
Again, all mentions of the flag need to be adjusted. It has to be automatic, or else the design will be error-prone and will cause either headaches or waste a lot of time reporting and investigating issues.

**People will miss this flag, trust me bro.** @thedrow

Reply:
People won't miss this flag because of the warning I print when using quorum queues without this feature enabled.

Reply:
So the way I see it,
Option 1) Log warning and expect user to see, read, understand, look for doc and adjust/fix code, even if it’s a simple line or two of change.
Option 2) No warning, automatic detection of state and automatic configuration accordingly. The user knows nothing, needs to configure nothing, and the behavior with/without quorum queues is the same from the POV the user/API, which is the definition of ETA support for quorum queues. Requiring to enable more flags creates a design dependency which affect the readability of the feature, breaks compatibility with the standard ETA feature (which doesn’t requires any flags) and by nature and definition, is error-prone (automatic/manual approach).

How come Option 2 doesn’t appeal to you @thedrow ?

Reply:
Because my client wants to test this feature without using quorum queues at first.
I think we can assume developers see this warning and that's just fine by me.

Reply:
Native delayed delivery is now enabled automatically when quorum queues are detected.